lib/client.js

```diff
@@ -251,14 +251,14 @@ Client.prototype.close = function (done) {
 
         function done (err) {
           if (!err) {
+            that.will = null // this function might be called twice
             that.broker.persistence.delWill({
               id: that.id,
               brokerId: that.broker.id
             }, nop)
           }
         }
       })
-      that.will = null // this function might be called twice
```

Comment:
Removing this line here allows the function to be called twice. You should leave this line, and make an in-closure variable for `that.will`.